raft/raft.go

```diff
@@ -890,8 +939,8 @@ func stepLeader(r *raft, m pb.Message) {
 	}
 
 	// All other message types require a progress for m.From (pr).
-	pr, prOk := r.prs[m.From]
-	if !prOk {
+	pr := r.getProgress(m.From)
```

Comment:
getProgress can return *p,ok, just like what the previous map struct does. it minimizes code changes. but not a big problem anyway.

Reply:
If here returns `*Progress, bool`, I will check the ok boolean in the places which originally use `getProgess().xxx` to replace `prs[id].xxx`, and this may cause code complex. 

Another way is to add a mustGetProgress function, but I think it is not necessary.